mvrx-mocking/src/main/kotlin/com/airbnb/mvrx/mocking/KotlinReflectUtils.kt

```diff
@@ -101,3 +106,40 @@ internal fun KClass<*>.findFunction(functionName: String): KFunction<*> {
     return functions.find { it.name == functionName }
         ?: error("No function found with name $functionName in class $simpleName")
 }
+
+/**
+ * Ensures that the given class is:
+ * - A kotlin data class
+ * - Contains no mutable properties
+ * - Contains no primary constructor params with a Mutable collection type.
+ *
+ * For further validations restricting usages of Java collections such as ArrayList, this can be combined with [com.airbnb.mvrx.assertMavericksDataClassImmutability]
+ */
+fun assertMavericksDataClassImmutabilityWithKotlinReflect(
+    kClass: KClass<*>,
+) {
+    require(kClass.isData) {
+        "Mavericks state must be a data class! - ${kClass::class.simpleName}"
+    }
+
+    kClass.primaryConstructor?.parameters?.forEach { param ->
+        when {
+            param.type.isSubtypeOf(MutableList::class.starProjectedType) -> {
```

Comment:
the function kdoc says `For further validations restricting usages of Java collections such as ArrayList, this can be combined with [com.airbnb.mvrx.assertMavericksDataClassImmutability]`. I wanted to avoid duplicate work and not do that here too (that other one is already automatically run, i viewed this as an additional optional check we could run if desired)